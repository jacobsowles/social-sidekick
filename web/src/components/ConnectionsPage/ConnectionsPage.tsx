import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

import ContentBox from '@components/ContentBox';
import LoadingSpinner from '@components/LoadingSpinner';
import PageHeader from '@components/PageHeader';
import ServiceIcon from '@components/ServiceIcon';
import { Service } from '@core/types';
import './ConnectionsPage.scss';

export interface ConnectionsPageProps {
  allServices: Service[];
  className?: string;
}

const ConnectionsPage: FunctionComponent<ConnectionsPageProps> = ({ allServices, className }) => {
  return (
    <main className={classNames('connections-page', className)}>
      <ContentBox>
        <PageHeader
          title="Connect Services"
          subtitle="Log into the services below to connect them to your Social Sidekick account."
        />
        {(!allServices || allServices.length === 0) && <LoadingSpinner />}
        {allServices &&
          allServices.map((service: Service, key: number) => (
            <ServiceIcon iconName={service.name.toLowerCase()} key={key} label={service.name} />
          ))}
      </ContentBox>
    </main>
  );
};

ConnectionsPage.defaultProps = {
  className: undefined
};

export default ConnectionsPage;
