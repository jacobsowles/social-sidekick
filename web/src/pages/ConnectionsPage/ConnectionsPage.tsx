import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

import ContentBox from '@components/ContentBox';
import LoadingSpinner from '@components/LoadingSpinner';
import PageHeader from '@components/PageHeader';
import ServiceIconContainer from '@components/ServiceIconContainer';
import { UserService } from '@core/types';
import './ConnectionsPage.scss';

export interface ConnectionsPageProps {
  className?: string;
  services: UserService[] | null;
  userId?: string;
}

const ConnectionsPage: FunctionComponent<ConnectionsPageProps> = ({ className, services }) => {
  return (
    <main className={classNames('connections-page', className)}>
      <ContentBox>
        <PageHeader
          title="Connect Services"
          subtitle="Log into the services below to connect them to your Social Sidekick account."
        />
        {(!services || services.length === 0) && <LoadingSpinner />}
        {services &&
          services.map((service: UserService) => (
            <ServiceIconContainer
              iconName={service.name.toLowerCase()}
              isConnected={service.isConnected}
              key={service._id}
              label={service.name}
              serviceId={service._id}
            />
          ))}
      </ContentBox>
    </main>
  );
};

ConnectionsPage.defaultProps = {
  className: undefined
};

export default ConnectionsPage;
