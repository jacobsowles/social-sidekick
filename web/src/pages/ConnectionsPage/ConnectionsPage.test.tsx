import React from 'react';

import { init, shallow, shallowTopLevelElement, toJson } from '@tests/component-test-base';
import ConnectionsPage, { ConnectionsPageProps } from './ConnectionsPage';

describe('ConnectionsPage', () => {
  let props: ConnectionsPageProps;
  init(() => <ConnectionsPage {...props} />, 'main');

  beforeEach(() => {
    props = {
      className: '',
      services: null
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });

  describe('when `className` is defined', () => {
    beforeEach(() => {
      props.className = 'another-class';
    });

    it('should include the specified class name', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.contain('another-class');
    });

    it('should include the connections-page class name', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.contain('connections-page');
    });
  });

  describe('when `className` is undefined', () => {
    it('should include the connections-page class name', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.contain('connections-page');
    });
  });
});
