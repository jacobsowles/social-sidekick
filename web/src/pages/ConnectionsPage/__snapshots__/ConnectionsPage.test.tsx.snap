// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionsPage should render the component without crashing 1`] = `
<main
  className="connections-page"
>
  <ContentBox>
    <PageHeader
      subtitle="Log into the services below to connect them to your Social Sidekick account."
      title="Connect Services"
    />
    <LoadingSpinner
      size="4x"
    />
  </ContentBox>
</main>
`;
